shared/src/main/scala/mlscript/NuTypeDefs.scala

```diff
@@ -985,6 +1060,54 @@ class NuTypeDefs extends ConstraintSolver { self: Typer =>
               
             case td: NuTypeDef =>
               
+              /** Check no `this` access in ctor statements or val rhs and reject unqualified accesses to virtual members.. */
+              def qualificationCheck(members: Ls[NuMember], stmts: Ls[Statement], baseAndSigs: Ls[NuMember]): Unit = {
+                val cache = mutable.HashMap[Str, Bool]()
+
+                def getMember(name: Str) =
+                  members.find(_.name === name).fold(baseAndSigs.find(_.name === name))(m => S(m))
```

Comment:
This is extremely inefficient. It seems you really like linear traversals! Why not create a map for fast retrieval?